invenio_records_rest/views.py

```diff
@@ -941,6 +953,60 @@ def check_etag(self, etag, weak=False):
                     )
 
 
+class SearchAsYouTypeResource(MethodView):
+    """Resource for `search_as_you_type` searches."""
+
+    view_name = '{0}_search_as_you_type'
+
+    def __init__(self, search_as_you_type, search_class=None, **kwargs):
+        """Constructor."""
+        self.search_as_you_type = search_as_you_type
+        self.search_as_you_type_fields = set(self.search_as_you_type.keys())
+        self.search_class = search_class
+
+    def get(self, **kwargs):
+        """Get results from `search_as_you_type`."""
+        request_fields = set(request.values.keys())
+        key = self.search_as_you_type_fields & request_fields
+
+        if not key:
+            raise SuggestNoCompletionsRESTError(
+                ', '.join(sorted(self.search_as_you_type.keys())))
+
+        key = key.pop()
+        value = request.values.get(key)
```

Comment:
What if it would be None?

Reply:
It will be an empty string because the value it's coming from the request values